### PR TITLE
CLI | Signing for the win binary was added to the CI (AST-42438)

### PR DESCRIPTION
**Pull request by Cx-DmitriyKhomenko**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

Added signing process for the windows binaries in the release CI

### References

https://checkmarx.atlassian.net/browse/AST-42438

### Testing

.

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).
- [x] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

File: .github/scripts/signing_win.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    exit 1
+fi
+
+# Sign
+ssh -n -i "$SSH_KEY_PATH" -o StrictHostKeyChecking=no "$REMOTE_USER@$REMOTE_HOST" "osslsigncode sign -certs /home/ubuntu/checkmarx.crt  -key 'pkcs11:object=CNGRSAPriv-cx-signing' -pass $HSM_CREDS -pkcs11module /opt/cloudhsm/lib/libcloudhsm_pkcs11.so -t http://timestamp.digicert.com -in '$REMOTE_PATH/$FILENAME' -out '$REMOTE_PATH/$FILENAME_SIGNED'"
+# Check remote command status
+if [ $? -ne 0 ]; then
+    echo "Failed to sign file $FILENAME"
+    ssh -n -i "$SSH_KEY_PATH" -o StrictHostKeyChecking=no "$REMOTE_USER@$REMOTE_HOST" "rm -f '$REMOTE_PATH/$FILENAME'"
+    exit 1
```
Comment (OrShamirCM) [inline .github/scripts/signing_win.sh:61]: @BenAlvo1 , please verify with Fatima if we want to brack the release when sign is failed

**Review by OrShamirCM — COMMENTED** [review]
Looks great!
should we maintain any secret? update once at X time?

**Comment by Cx-DmitriyKhomenko** [comment]
> Looks great! should we maintain any secret? update once at X time?

@OrShamirCM Nothing is required on your side. The code signing certificate is updated once a year, but it is not listed anywhere in the secrets and does not require any updates from you.

**Review by AlvoBen — APPROVED** [review]

File: .github/scripts/signing_win.sh (hunk 1 above)
Comment (AlvoBen) [reply .github/scripts/signing_win.sh:61]: I added this functionality

File: .goreleaser-dev.yml [hunk 2]
```diff
@@ -32,6 +32,16 @@ builds:
       - -s
       - -w
       - -X github.com/checkmarx/ast-cli/internal/params.Version={{.Version}}
+    hooks:
+      post:
+        - cmd: bash .github/scripts/signing_win.sh dist/cx_windows_amd64_v1/cx.exe {{.Os}} || true
+          output: true
```
Comment (OrShamirCM) [inline .goreleaser-dev.yml:38]: is this the check for dev? if you are exit with 1 it'll not fail everything?

Comment (AlvoBen) [reply .goreleaser-dev.yml:38]: No because the condition 'or true' will cause this line to continue with the process without failing the pipeline

**Review by OrShamirCM — APPROVED** [review]

**Pull request merged**